pkg/reconciler/managed/reconciler.go

```diff
@@ -1096,28 +1115,123 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	return reconcile.Result{RequeueAfter: r.pollInterval}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 }
 
+// isManagementPolicySupported returns true if the management policy is
+// in the list of supported management policies. Short-circuits to true if
+// management policies are disabled.
+func isManagementPolicySupported(enabled bool, policy xpv1.ManagementPolicy) bool {
+	if !enabled {
+		return true
+	}
+	policySet := sets.New[xpv1.ManagementAction](policy...)
+	for _, p := range supportedManagementPolicies {
+		if p.Equal(policySet) {
+			return true
+		}
+	}
+	return false
+}
+
+// ContainsManagementAction returns true if the management policy contains
+// the management action or ManagementActionAll.
+func ContainsManagementAction(managementPolicy xpv1.ManagementPolicy, action xpv1.ManagementAction) bool {
```

Comment:
Any reason why some of these are public and others private?

Reply:
These ones are the most generic and used by the providers (at least I used it in the [upjet PR](https://github.com/upbound/upjet/pull/224)). For the others, it seemed to me that they are very specific to the reconciler so I left them private. 

They will probably change/go away as I work on [the comment below](https://github.com/crossplane/crossplane-runtime/pull/456#discussion_r1259102918).